### PR TITLE
Create MusicAlbum 

### PR DESCRIPTION
**Pull request by tresorsawasawa**
In this PR, I completed the following tasks:
- Create MusicAlbum class in a separate .rb file
- Create a Genre class with an association to the Item class (in a separate .rb file).
- All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- add_item method in the Genre class
   - should take an instance of the Item class as an input
   - should add the input item to the collection of items
   - should add self as a property of the item object (by using the correct setter from the item object)
- can_be_archived?() in the MusicAlbum class
   - should override the method from the parent class
   - should return true if the parent's method returns true AND if on_spotify equals true
   - otherwise, it should return false

### CODE REVIEW

File: classes/genre.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require_relative 'item'
+
+class Genre
+  attr_accessor :name
+  attr_reader :id, :items
+
+  def initialize(name)
+    @id = id
```
Comment (ahmedtaa) [inline classes/genre.rb:8]: you should asign a value to @id not just id, use rand(1...1000)

File: classes/genre.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require_relative 'item'
+
+class Genre
+  attr_accessor :name
+  attr_reader :id, :items
```
Comment (ahmedtaa) [inline classes/genre.rb:5]: id should be moved to arre_accessor because we want to to set it.

File: classes/music_album.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(on_spotify, publish_date, archived)
+    super(publish_date, archived)
+    @on_spotify = on_spotify
+  end
+
+  private
+
+  def can_be_archived
+    return true if super && @on_spotify
+  end
```
Comment (ahmedtaa) [inline classes/music_album.rb:16]: should return false in case the return is not true
just add false before the end keyword

**Review by ahmedtaa — CHANGES_REQUESTED** [review]

**Review by ahmedtaa — CHANGES_REQUESTED** [review]
Change Required 🛑 
Hi, @tresorsawasawa Nice work so far.
Please check my comments

File: classes/genre.rb (hunk 1 above)
Comment (tresorsawasawa) [reply classes/genre.rb]: Thank You. I am sorry for the error

File: classes/genre.rb (hunk 2 above)
Comment (tresorsawasawa) [reply classes/genre.rb:5]: Thank you

**Review by Gegardus — APPROVED** [review]
Good job. Thank you for implementing previous reviewer changes. It's time to merge it.

File: classes/music_album.rb (hunk 3 above)
Comment (ahmedtaa) [reply classes/music_album.rb:16]: Hi @tresorsawasawa 
The method should look like this
```
def can_be_archived
    return true if super && @on_spotify

    false
  end
```

**Pull request merged**